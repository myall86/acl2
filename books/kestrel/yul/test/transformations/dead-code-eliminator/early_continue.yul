{
    let a := 20
    for {
    }
    mylt(a, 40)
    {
        a := myadd(a, 2)
    }
    {
        a := a
        continue
        mymstore(0, a)
        a := myadd(a, 10)
    }
    function mylt(a, b) -> c { }
    function myadd(a,b) -> c { }
    function mymstore(a,b) { }
}

// ----
// step: deadCodeEliminator
//
// {
//     let a := 20
//     for { } mylt(a, 40) { a := myadd(a, 2) }
//     {
//         a := a
//         continue
//     }
//     function mylt(a, b) -> c { }
//     function myadd(a,b) -> c { }
//     function mymstore(a,b) { }
// }
