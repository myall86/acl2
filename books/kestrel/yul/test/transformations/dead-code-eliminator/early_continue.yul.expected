
 {
     let a := 20
     for { } mylt(a, 40) { a := myadd(a, 2) }
     {
         a := a
         continue
     }
     function mylt(a, b) -> c { }
     function myadd(a,b) -> c { }
     function mymstore(a,b) { }
 }
